web/src/components/DandisetsPage.vue

```diff
@@ -55,48 +76,86 @@ export default {
       type: String,
       required: true,
     },
-    module: {
-      type: String,
-      required: true,
+    user: {
+      type: Boolean,
+      required: false,
+      default: false,
     },
   },
   data() {
     return {
-      selection: 'newest',
-      options: basicDandisetSortingOptions,
+      sortingOptions,
+      sortOption: Number(this.$route.query.sortOption) || 0,
+      sortDir: this.$route.query.sortDir || 1,
+      page: Number(this.$route.query.page) || 1,
+      total: 0,
     };
   },
   computed: {
-    page: {
-      get() {
-        return this.moduleRoot.page;
-      },
-      set(page) {
-        this.changePage({ page });
-      },
+    listingUrl() {
+      return this.user ? 'dandi/user' : 'dandi';
     },
-    moduleRoot() {
-      return this.$store.state[this.module];
+    pages() {
+      return Math.ceil(this.total / DANDISETS_PER_PAGE) || 1;
     },
-    dandisets() {
-      return this.moduleRoot.dandisets;
+    sortField() {
+      return this.sortingOptions[this.sortOption].field;
     },
-    pages() {
-      return this.moduleRoot.pages;
+    queryParams() {
+      const {
+        page, sortOption, sortDir,
+      } = this;
+
+      return {
+        page,
+        sortOption,
+        sortDir,
+      };
+    },
+  },
+  asyncComputed: {
+    async dandisets() {
+      const {
+        listingUrl, page, sortField, sortDir,
+      } = this;
+
+      const { data: dandisets } = await girderRest.get(listingUrl, {
+        params: {
+          limit: DANDISETS_PER_PAGE,
+          offset: (page - 1) * DANDISETS_PER_PAGE,
+          sort: sortField,
+          sortdir: sortDir,
+        },
+      });
+
+      return dandisets;
+    },
+  },
+  watch: {
+    queryParams(query) {
+      this.$router.replace({
+        ...this.$route,
+        query,
+      });
     },
   },
   created() {
-    this.reload();
+    this.getTotalDandisets();
   },
   methods: {
-    changeSort(payload) {
-      return this.$store.dispatch(`${this.module}/changeSort`, payload);
-    },
-    changePage(payload) {
-      return this.$store.dispatch(`${this.module}/changePage`, payload);
+    changeSort(index) {
+      if (this.sortOption === index) {
+        this.sortDir *= -1;
+      } else {
+        this.sortOption = index;
+        this.sortDir = 1;
+      }
+
+      this.page = 1;
     },
-    reload(payload) {
-      return this.$store.dispatch(`${this.module}/reload`, payload);
+    async getTotalDandisets() {
+      const { headers: { 'girder-total-count': total } } = await girderRest.get(this.listingUrl, { params: { limit: 1 } });
```

Comment:
I'm not a huge fan of this extra request every time the component is mounted. What do you think of something like this:
```
asyncComputed: {
  async request() {
    return await girderRest.get(listingUrl, ...);
  },
  computed: {
    dandisets() {
      return this.request.data;
    },
    totalDandisets() {
      return request.headers['girder-total-count'];
    },
  }
}
```
It feels pretty janky to have the whole request object stored like that, but I feel bad making an extra request just to get a number that is already provided on every request.

Reply:
I like this idea, I'll be pushing this change soon.